src/runtime_src/core/edge/drm/zocl/zocl_kds.c

```diff
@@ -341,23 +345,31 @@ int zocl_del_context_kernel(struct drm_zocl_dev *zdev, void *client_hdl, u32 cu_
 
 	mutex_lock(&client->lock);
 	cctx = zocl_check_exists_context(client, &uuid_null);
-	if (cctx == NULL)
+	if (cctx == NULL) {
+		mutex_unlock(&client->lock);
 		return -EINVAL;
+	}
 
```

Comment:
I have a question about this code. It seems that the cctx is allocated from the zocl_check_exists_context and should be freed before returning this call. But there arem many places that we don't free the cctx. Is that expected behavior?